### ISSUE TITLE
Clear interpreter warnings

### ISSUE THREAD

**Issue by jdantonio** [ruby-concurrency/concurrent-ruby#417]
When running the Rails test suit against the latest pre-release we see a bunch of interpreter warnings. We need to clear as many of these as possible before the next release. Here is the full output:

```
[10:06:06 jerry ~/Projects/rails/actionmailer (replace-thread-safe)]$ rake test
/home/jerry/.rvm/rubies/ruby-2.2.2/bin/ruby -w -I"lib:test" -I"/home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0" "/home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb" "test/**/*_test.rb" 
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/utility/native_extension_loader.rb:1: warning: loading in progress, circular require considered harmful - /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization.rb
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in  `<main>'
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in  `select'
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in  `block in <main>'
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in  `each'
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in  `block (2 levels) in <main>'
	from /home/jerry/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in  `require'
	from /home/jerry/Projects/rails/actionmailer/test/assert_select_email_test.rb:1:in  `<top (required)>'
	from /home/jerry/Projects/rails/actionmailer/test/assert_select_email_test.rb:1:in  `require'
	from /home/jerry/Projects/rails/actionmailer/test/abstract_unit.rb:13:in  `<top (required)>'
	from /home/jerry/Projects/rails/actionmailer/test/abstract_unit.rb:13:in  `require'
	from /home/jerry/Projects/rails/actionmailer/lib/action_mailer.rb:24:in  `<top (required)>'
	from /home/jerry/Projects/rails/actionmailer/lib/action_mailer.rb:24:in  `require'
	from /home/jerry/Projects/rails/actionpack/lib/abstract_controller.rb:2:in  `<top (required)>'
	from /home/jerry/Projects/rails/actionpack/lib/abstract_controller.rb:2:in  `require'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/rails.rb:15:in  `<top (required)>'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/rails.rb:15:in  `require'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/dependencies/autoload.rb:1:in  `<top (required)>'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/dependencies/autoload.rb:1:in  `require'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflector/methods.rb:1:in  `<top (required)>'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflector/methods.rb:1:in  `require'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflections.rb:1:in  `<top (required)>'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflections.rb:1:in  `require'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflector/inflections.rb:1:in  `<top (required)>'
	from /home/jerry/Projects/rails/activesupport/lib/active_support/inflector/inflections.rb:1:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent.rb:2:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent.rb:2:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/configuration.rb:2:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/configuration.rb:2:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/delay.rb:2:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/delay.rb:2:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/concern/obligation.rb:4:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/concern/obligation.rb:4:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/atomic/event.rb:2:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/atomic/event.rb:2:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization.rb:14:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization.rb:14:in  `require'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/utility/native_extension_loader.rb:1:in  `<top (required)>'
	from /home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/utility/native_extension_loader.rb:1:in  `require'
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/atom.rb:64: warning: `*' interpreted as argument prefix
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:82: warning: instance variable @safe_initialization not initialized
/home/jerry/.rvm/gems/ruby-2.2.2@rails/gems/concurrent-ruby-1.0.0.pre2/lib/concurrent/synchronization/object.rb:125: warning: `&' interpreted as argument prefix
Run options: --seed 44955

# Running:

.....................................................................................................................................................................

Finished in 0.676516s, 243.8965 runs/s, 548.3976 assertions/s.

165 runs, 371 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by jdantonio** [comment]
Rails [PR 21679](https://github.com/rails/rails/pull/21679).

@pitr-ch Some of these warnings come from the latest updates to the synchronization layer. Would you like to look into those, or would you like me to?

**Comment by pitr-ch** [comment]
@jdantonio heh, just pulled latest changes to find out your fix conflicts with my fix :) Thanks for the fix.

**Comment by jdantonio** [comment]
I haven't merged this yet so we can close this and use yours.

**Comment by pitr-ch** [comment]
They are already in master, the fixes where from @amatsuda, thanks!

**Comment by jdantonio** [comment]
@pitr-ch Ah. I got my PRs confused. The changes that were merged into master were submitted by a new contributor who saw the Rails PR and jumped in to help.

**Comment by jdantonio** [comment]
Fixed in 1.0.0.pre4
